include/CXXGraph/Graph/Graph_decl.h

```diff
@@ -35,6 +35,7 @@
 #include <unordered_map>
 #include <unordered_set>
 #include <utility>
+#include <algorithm>
```

Comment:
Why introduce `<algorithm>` in `Graph_decl.h`? Maybe is better introduce it when it needed